flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -4,247 +4,278 @@ import UIKit
+
+            self.sentryOptions = options
+
+            // note : for now, in sentry-cocoa, beforeSend is not called before captureEnvelope
+            options.beforeSend = { event in
+                self.setEventOriginTag(event: event)
+
+                if var sdk = event.sdk, self.isValidSdk(sdk: sdk) {
+                    if let packages = arguments["packages"] as? [String] {
+                        if var sdkPackages = sdk["packages"] as? [String] {
+                            sdk["packages"] = sdkPackages.append(contentsOf: packages)
+                        } else {
+                            sdk["packages"] = packages
+                        }
+                    }
+
+                    if let integrations = arguments["integrations"] as? [String] {
+                        if var sdkIntegrations = sdk["integrations"] as? [String] {
+                            sdk["integrations"] = sdkIntegrations.append(contentsOf: integrations)
+                        } else {
+                            sdk["integrations"] = integrations
+                        }
+                    }
+                    event.sdk = sdk
+                }
+
+                return event
             }
-          }
-          event.sdk = sdk
         }
 
-        return event
-      }
+        if startSession && sentryOptions?.enableAutoSessionTracking == true {
+            // only this is not enough, we need to mimics SentrySessionTracker.didBecomeActive
+            // how to call that from here?
+            // it'd also be nice to know if theres already a running session, so we dont do twice, lets say that
+            // not always the SDK is init. too late.
+            SentrySDK.currentHub().startSession()
+        }
+
+        result("")
     }
 
-    result("")
-  }
+    // swiftlint:disable:next cyclomatic_complexity
+    private func updateOptions(arguments: [String: Any], options: Options) {
+        if let dsn = arguments["dsn"] as? String {
+            options.dsn = dsn
+        }
 
-  // swiftlint:disable:next cyclomatic_complexity
-  private func updateOptions(arguments: [String: Any], options: Options) {
-    if let dsn = arguments["dsn"] as? String {
-      options.dsn = dsn
-    }
+        if let isDebug = arguments["debug"] as? Bool {
+            options.debug = isDebug
+        }
 
-    if let isDebug = arguments["debug"] as? Bool {
-      options.debug = isDebug
-    }
+        if let environment = arguments["environment"] as? String {
+            options.environment = environment
+        }
 
-    if let environment = arguments["environment"] as? String {
-      options.environment = environment
-    }
+        if let releaseName = arguments["release"] as? String {
+            options.releaseName = releaseName
+        }
 
-    if let releaseName = arguments["release"] as? String {
-      options.releaseName = releaseName
-    }
+        if let enableAutoSessionTracking = arguments["enableAutoSessionTracking"] as? Bool {
```

Comment:
Unrelated to this change but cocoa has an overload that takes a map for this use case (it's used by React Native).

Reply:
it'd work only if all fields match, its not the case right now, but likely to be addressed once https://github.com/getsentry/sentry-java/issues/1117 is done